### PR TITLE
feat: scaffold yeoman generator

### PR DESCRIPTION
**Pull request by naomiaro**
Bootstraps a yeoman generator which scaffolds files related to asdf tool installations:

```
.tool-versions
requirements.txt
Makefile
```

Prompts are collected for asdf tool versions.

### CODE REVIEW

File: generator-cas-app/LICENSE [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Copyright 2021 Naomi Aro <naomi.aro@gov.bc.ca>
```
Comment (matthieu-foucault) [inline generator-cas-app/LICENSE:1]: maybe copy the LICENSE file that is at the root of this project? It should probably be copyrighted with "Government of British Columbia <ggircs@gov.bc.ca>" (the license file at the root should be updated too as it has a placeholder year and owner, it is probably the same in all of our repos :upside_down_face: )

File: generator-cas-app/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# generator-cas-app
+> Generates various scaffolds for a new CAS application.
+
+## Installation
+
+```bash
+npm install -g yo
+npm link
```
Comment (matthieu-foucault) [inline generator-cas-app/README.md:8]: use yarn instead?

File: generator-cas-app/generators/app/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        type: "input",
+        name: "nodejs",
+        message: "node.js version",
+        default: "14.17.6"
+      },
+      {
+        type: "input",
+        name: "yarn",
+        message: "yarn version",
+        default: "1.22.5"
```
Comment (matthieu-foucault) [inline generator-cas-app/generators/app/index.js:29]: This should probably default to `1.22.10`
Something we discovered while starting a new project at Button is that the asdf yarn plugin only supports yarn 1. Yarn 2 or 3 are managed differently. Not sure if this is something we want to handle. If we wanted to support yarn 2+, the solution is to have `yarn 1.22.10` in the `.tool-versions` and use the `yarn set version` command (https://yarnpkg.com/cli/set/version)

File: generator-cas-app/generators/app/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      this.templatePath("requirements.txt"),
+      this.destinationPath("requirements.txt")
+    );
+
+    this.fs.copy(
+      this.templatePath("Makefile"),
+      this.destinationPath("Makefile")
+    );
+  }
+};
```
Comment (matthieu-foucault) [inline generator-cas-app/generators/app/index.js:65]: Can yeoman run `make install_asdf_tools` in the new directory once it is done copying things?

File: generator-cas-app/package-lock.json [hunk 5]
```diff
@@ -0,0 +1,9474 @@
+{
```
Comment (matthieu-foucault) [inline generator-cas-app/package-lock.json:1]: let's use yarn instead of npm

File: generator-cas-app/generators/app/templates/.tool-versions [hunk 6]
```diff
@@ -0,0 +1,4 @@
+nodejs <%= nodejs %>
```
Comment (matthieu-foucault) [inline generator-cas-app/generators/app/templates/.tool-versions:1]: should all of our template files have the `.ejs` extension?

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: generator-cas-app/README.md (hunk 2 above)
Comment (naomiaro) [reply generator-cas-app/README.md:8]: The thing with yarn at this point, is that asdf hasn't been run yet, so yarn will have to be installed another way.

Comment (naomiaro) [reply generator-cas-app/README.md:8]: But it's also the same thing with node as well... 🤷🏻

File: generator-cas-app/generators/app/index.js (hunk 4 above)
Comment (naomiaro) [reply generator-cas-app/generators/app/index.js:65]: I think so.., I'll try!

File: generator-cas-app/generators/app/templates/.tool-versions (hunk 6 above)
Comment (naomiaro) [reply generator-cas-app/generators/app/templates/.tool-versions:1]: No it doesn't need to.

File: generator-cas-app/LICENSE (hunk 1 above)
Comment (naomiaro) [reply generator-cas-app/LICENSE:1]: This shorter one is maybe the license that should be applied at both levels? http://www.apache.org/licenses/LICENSE-2.0#apply It was auto generated by the yeoman scaffold like this, but I should swap my name out for sure.

File: generator-cas-app/generators/app/templates/Makefile [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+	@asdf plugin-update --all
+	@bash ~/.asdf/plugins/nodejs/bin/import-release-team-keyring
+	@#MAKELEVEL=0 is required because of https://www.postgresql.org/message-id/1118.1538056039%40sss.pgh.pa.us
+	@MAKELEVEL=0 POSTGRES_EXTRA_CONFIGURE_OPTIONS='--with-libxml' asdf install
+	@asdf reshim
+	@pip install -r requirements.txt
+	@asdf reshim
+
+.PHONY: install_dev_tools
+install_dev_tools: $(call make_help,install_dev_tools,install development tools via asdf)
```
Comment (matthieu-foucault) [inline generator-cas-app/generators/app/templates/Makefile:15]: This `make_help` function requires some functions defined in the `cas-pipeline` repo. I however found a much easier solution so make a `make help` command work:
1) add the following target at the top of the file (under the SHELL line):
```makefile
help:     ## Show this help.
	@sed -ne '/@sed/!s/## //p' $(MAKEFILE_LIST)
```

and replace the line above with this:
```suggestion
install_dev_tools: ## install development tools via asdf
```

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
just one small change and it's good to go

File: generator-cas-app/generators/app/templates/Makefile (hunk 7 above)
Comment (naomiaro) [reply generator-cas-app/generators/app/templates/Makefile:15]: That makes sense, I was wondering what was going on here, not a Makefile expert 😀

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**